Fixes #56: Landscape layouts

ShankhanilSaha opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

Fix #56 
- Removes portrait lock from `AndroidManifest.xml` files.
- Removes related TODO comments.
- Uncomments related tests.
- Creates tests `testStateFragment_configurationChange_previousButtonStateRetained()` and `testStateFragment_configurationChange_nextButtonStateRetained()
` in `app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt`.
-  Removed lint exemption for Orientation Locks in `android_lint_exemptions.textproto` as all orientation locks in the codebase have been removed and those are no longer required.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

ShankhanilSaha committed:
remove orientation lock from manifest

ShankhanilSaha committed:
remove ignore annotations from tests

ShankhanilSaha committed:
add previous, next button state retention checks

ShankhanilSaha committed:
remove path configurations

ShankhanilSaha committed:
remove TODO comment from manifest

theayushyadav11 commented:
> @theayushyadav11 one of the tests mentioned in task 2 fails, I will be raising a PR of my changes as i have made all the mentioned required changes?

Hi @ShankhanilSaha , can you please explicitly mention which test is failing and share the error logs.
These would help in locating the actual  cause.
Thnaks.

ShankhanilSaha commented:
> > @theayushyadav11 one of the tests mentioned in task 2 fails, I will be raising a PR of my changes as i have made all the mentioned required changes?
> 
> Hi @ShankhanilSaha , can you please explicitly mention which test is failing and share the error logs. These would help in locating the actual cause. Thnaks.

Hi there @theayushyadav11 , here are the logs for the failed test, the following test was located in the `app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest` file.
`Time: 10.272
There was 1 failure:
1) testAudioFragment_invokePrepared_playAudio_configurationChange_checkStillPlaying(org.oppia.android.app.player.audio.AudioFragmentTest)
java.lang.IllegalArgumentException: Don't know what to do with dataSource org.robolectric.shadows.util.DataSource@1d2c40f9 - either add an exception with addException() or media info with addMediaInfo()
        at org.robolectric.shadows.ShadowMediaPlayer.doSetDataSource(ShadowMediaPlayer.java:643)
        at org.robolectric.shadows.ShadowMediaPlayer.setDataSource(ShadowMediaPlayer.java:581)
        at org.robolectric.shadows.ShadowMediaPlayer.setDataSource(ShadowMediaPlayer.java:587)
        at android.media.MediaPlayer.setDataSource(MediaPlayer.java)
        at org.oppia.android.domain.audio.AudioPlayerController.prepareDataSource(AudioPlayerController.kt:153)
        at org.oppia.android.domain.audio.AudioPlayerController.changeDataSource(AudioPlayerController.kt:122)
        at org.oppia.android.app.player.audio.AudioViewModel.setAudioLanguageCode(AudioViewModel.kt:135)
        at org.oppia.android.app.player.audio.AudioViewModel.loadAudio(AudioViewModel.kt:116)
        at org.oppia.android.app.player.audio.AudioViewModel.loadMainContentAudio(AudioViewModel.kt:77)
        at org.oppia.android.app.player.audio.AudioFragmentPresenter.loadMainContentAudio(AudioFragmentPresenter.kt:225)
        at org.oppia.android.app.player.audio.AudioFragment.loadMainContentAudio(AudioFragment.kt:91)
        at org.oppia.android.app.testing.AudioFragmentTestActivityPresenter.handleOnCreate(AudioFragmentTestActivityPresenter.kt:51)
        at org.oppia.android.app.testing.AudioFragmentTestActivity.onCreate(AudioFragmentTestActivity.kt:24)
        at android.app.Activity.performCreate(Activity.java:7995)
        at android.app.Activity.performCreate(Activity.java:7979)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
        at org.robolectric.android.internal.RoboMonitoringInstrumentation.callActivityOnCreate(RoboMonitoringInstrumentation.java:284)
        at org.robolectric.android.controller.ActivityController.lambda$create$0(ActivityController.java:113)
        at org.robolectric.shadows.ShadowPausedLooper.runPaused(ShadowPausedLooper.java:203)
        at org.robolectric.android.controller.ActivityController.create(ActivityController.java:113)
        at org.robolectric.android.controller.ActivityController.create(ActivityController.java:118)
        at org.robolectric.android.internal.RoboMonitoringInstrumentation.startActivitySyncInternal(RoboMonitoringInstrumentation.java:103)
        at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:33)
        at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:38)
        at androidx.test.core.app.ActivityScenario.launchInternal(ActivityScenario.java:272)
        at androidx.test.core.app.ActivityScenario.launch(ActivityScenario.java:229)
        at org.oppia.android.app.player.audio.AudioFragmentTest.testAudioFragment_invokePrepared_playAudio_configurationChange_checkStillPlaying(AudioFragmentTest.kt:363)
        Suppressed: org.robolectric.android.internal.AndroidTestEnvironment$UnExecutedRunnablesException: Main looper has queued unexecuted runnables. This might be the cause of the test failure. You might need a shadowOf(Looper.getMainLooper()).idle() call.

FAILURES!!!
Tests run: 10,  Failures: 1`

ShankhanilSaha commented:
> > > @theayushyadav11 one of the tests mentioned in task 2 fails, I will be raising a PR of my changes as i have made all the mentioned required changes?
> > 
> > 
> > Hi @ShankhanilSaha , can you please explicitly mention which test is failing and share the error logs. These would help in locating the actual cause. Thnaks.
> 
 >Hi there @theayushyadav11 , here are the logs for the failed test, the following test was located in the >`app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest` file. 
>`Time: 10.272 There was 1 failure:
> 
> 1. testAudioFragment_invokePrepared_playAudio_configurationChange_checkStillPlaying(org.oppia.android.app.player.audio.AudioFragmentTest)
>    java.lang.IllegalArgumentException: Don't know what to do with dataSource org.robolectric.shadows.util.DataSource@1d2c40f9 - either add an exception with addException() or media info with addMediaInfo()
>    at org.robolectric.shadows.ShadowMediaPlayer.doSetDataSource(ShadowMediaPlayer.java:643)
>    at org.robolectric.shadows.ShadowMediaPlayer.setDataSource(ShadowMediaPlayer.java:581)
>    at org.robolectric.shadows.ShadowMediaPlayer.setDataSource(ShadowMediaPlayer.java:587)
>    at android.media.MediaPlayer.setDataSource(MediaPlayer.java)
>    at org.oppia.android.domain.audio.AudioPlayerController.prepareDataSource(AudioPlayerController.kt:153)
>    at org.oppia.android.domain.audio.AudioPlayerController.changeDataSource(AudioPlayerController.kt:122)
>    at org.oppia.android.app.player.audio.AudioViewModel.setAudioLanguageCode(AudioViewModel.kt:135)
>    at org.oppia.android.app.player.audio.AudioViewModel.loadAudio(AudioViewModel.kt:116)
>    at org.oppia.android.app.player.audio.AudioViewModel.loadMainContentAudio(AudioViewModel.kt:77)
>    at org.oppia.android.app.player.audio.AudioFragmentPresenter.loadMainContentAudio(AudioFragmentPresenter.kt:225)
>    at org.oppia.android.app.player.audio.AudioFragment.loadMainContentAudio(AudioFragment.kt:91)
>    at org.oppia.android.app.testing.AudioFragmentTestActivityPresenter.handleOnCreate(AudioFragmentTestActivityPresenter.kt:51)
>    at org.oppia.android.app.testing.AudioFragmentTestActivity.onCreate(AudioFragmentTestActivity.kt:24)
>    at android.app.Activity.performCreate(Activity.java:7995)
>    at android.app.Activity.performCreate(Activity.java:7979)
>    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
>    at org.robolectric.android.internal.RoboMonitoringInstrumentation.callActivityOnCreate(RoboMonitoringInstrumentation.java:284)
>    at org.robolectric.android.controller.ActivityController.lambda$create$0(ActivityController.java:113)
>    at org.robolectric.shadows.ShadowPausedLooper.runPaused(ShadowPausedLooper.java:203)
>    at org.robolectric.android.controller.ActivityController.create(ActivityController.java:113)
>    at org.robolectric.android.controller.ActivityController.create(ActivityController.java:118)
>    at org.robolectric.android.internal.RoboMonitoringInstrumentation.startActivitySyncInternal(RoboMonitoringInstrumentation.java:103)
>    at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:33)
>    at org.robolectric.android.internal.LocalActivityInvoker.startActivity(LocalActivityInvoker.java:38)
>    at androidx.test.core.app.ActivityScenario.launchInternal(ActivityScenario.java:272)
>    at androidx.test.core.app.ActivityScenario.launch(ActivityScenario.java:229)
>    at org.oppia.android.app.player.audio.AudioFragmentTest.testAudioFragment_invokePrepared_playAudio_configurationChange_checkStillPlaying(AudioFragmentTest.kt:363)
>    Suppressed: org.robolectric.android.internal.AndroidTestEnvironment$UnExecutedRunnablesException: Main looper has queued unexecuted runnables. This might be the cause of the test failure. You might need a shadowOf(Looper.getMainLooper()).idle() call.
> 
> FAILURES!!! Tests run: 10, Failures: 1`

From what i understand seems like the test fails not due to a issue in the app but due to how the media player is being handled in the test itself.

adhiamboperes commented:
@ShankhanilSaha, this line in the logs: 

>    java.lang.IllegalArgumentException: Don't know what to do with dataSource org.robolectric.shadows.util.DataSource@1d2c40f9 - either add an exception with addException() or media info with addMediaInfo()

Let's us know what is wrong with the test. In this test suite, there is a helper function, `addMediaInfo()`. From checking it's usages, it is meant to be called at the top of the test before launching the activity. Please see `testAudioFragment_invokePrepared_clickPlay_touchSeekBar_checkStillPlaying`. This means you will need to remove ` invokePreparedListener(shadowMediaPlayer)` from the test.

Additionally, you are likely to experience some errors due to some views not being visible. Please ensure `testCoroutineDispatchers.runCurrent()` is called inside the `use` block, before any view actions. This allows the activity to fully load and the test to reach an idle state. `testCoroutineDispatchers.runCurrent()`  is also typically used after every `perform` action, for the same reasons.

ShankhanilSaha committed:
fix failing test's setup

ShankhanilSaha commented:
Hi @adhiamboperes , i have implemented the helper function and removed the `invokePreparedListener(shadowMediaPlayer)` call from the test. It seems to be working fine now.

adhiamboperes commented:
That's good to hear, @ShankhanilSaha! I have approved the CI runs. If all the tests pass, please let me know so that I can proceed with the review.

ShankhanilSaha committed:
remove locked orientation exemption

ShankhanilSaha commented:
Hi there @adhiamboperes, most of the tests passed except a lint check which was caused by a exemption in `android_lint_exemptions.textproto`  for orientation locks, since the locks are now removed it caused  the test to fail. Should work now as I have removed it.

adhiamboperes commented on instrumentation/src/java/AndroidManifest.xml:
I searched this file and there are 5 instances of `android:screenOrientation="portrait"`, that need to be removed.

adhiamboperes replied:
It would be useful to do a global search in the project and verify that all instances have been removed.

adhiamboperes reviewed (changes requested):
Great job @ShankhanilSaha! I just have one follow up. PTAL.

adhiamboperes commented:
Also, per the lint check results, you need to delete the locked orientation enum field.

ShankhanilSaha committed:
remove portrait mode locks and unused enum fields

ShankhanilSaha reviewed (commented):
I have taken care of the locks.

ShankhanilSaha commented:
Also the unused enum fields are deleted.
@adhiamboperes

adhiamboperes reviewed (approved):
LGTM, thanks @ShankhanilSaha!
